tracing-futures/src/lib.rs

```diff
@@ -302,6 +303,63 @@ impl<T: Sink> Sink for Instrumented<T> {
     }
 }
 
+#[cfg(all(feature = "futures-03", feature = "std-future"))]
+#[cfg_attr(docsrs, doc(cfg(all(feature = "futures-03", feature = "std-future"))))]
+impl<T: futures::Stream> futures::Stream for Instrumented<T> {
+    type Item = T::Item;
+
+    fn poll_next(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> futures::task::Poll<Option<Self::Item>> {
+        let this = self.project();
+        let _enter = this.span.enter();
+        T::poll_next(this.inner, cx)
+    }
+}
+
+#[cfg(all(feature = "futures-03", feature = "std-future"))]
```

Comment:
I think we'll probably eventually want to just get rid of the `std-future` feature and make it always-on, now that it's supported on our minimum Rust version. But, we should do that in a separate PR.